firedrake/meshadapt.py

```diff
@@ -0,0 +1,113 @@
+from __future__ import absolute_import, print_function, division
+from six import with_metaclass
+import abc
+
+import numpy as np
+
+from firedrake.mesh import Mesh
+import firedrake.dmplex as dmplex
+import firedrake.utils as utils
+import firedrake.functionspace as functionspace
+import firedrake.function as function
+import firedrake.mesh as fmesh
+
+
+__all__ = ['adapt']
+
+
+class BaseAdaptation(with_metaclass(abc.ABCMeta)):
```

Comment:
FWIW, I prefer if the bikeshed is painted `BaseAdaptor` and `AnisotropicAdaptor`, rather than `BaseAdaptation` and `AnisotropicAdaptation`.